src/oracle/OracleWriter.sol

```diff
@@ -1,16 +1,56 @@
 pragma solidity 0.8.13;
 
-import "src/interfaces/IOracleWriter.sol";
+import { IOracleWriter } from "src/interfaces/IOracleWriter.sol";
+import { LogCompression } from "src/libraries/LogCompression.sol";
+import { stdMath } from "forge-std/Test.sol";
 
 abstract contract OracleWriter is IOracleWriter {
+    // 10 basis points per second which is 6% per minute and doubling of the price in 16 minutes
+    uint256 internal constant MAX_CHANGE_PER_SEC = 0.001e18;
+
     Observation[65536] public observations;
     uint16 public index = type(uint16).max;
 
+    // maximum allowed rate of change of price per second
+    // for clamping the price
+    uint64 internal allowedChangePerSecond = 0.0005e18; // 5 bp
```

Comment:
Feels arbitrary, shouldn't we just have a constructor argument?

Reply:
if we have a constructor argument, who provides this rate? Certainly not the user? If it's not the user 

My thinking was we can set a sensible rate as the default value which all pairs use. Then in production, for specific pairs that will be used as price feeds but have too loose / tight a clamp (which is probably not many), we can then tweak it

Reply:
`Reading from the factory default config is not so diff from the default I'm value I'm using here.` - The factory default can be edited over time so its completely different. This one is immutable.

Reply:
okay makes sense. Made it read from the factory on instantiation. 

Also, since `setMaxChangePerSecond` and the constructor now need to know about the factory (previously they didn't), we have to change the inheritance a bit (make `OracleWriter` inherit from `Pair`), so that the impl of `setAllowedChangePerSecond` can stay inside `OracleWriter` instead of having to be in `ReservoirPair`.

Reply:
okay